signer/src/transaction_coordinator.rs

```diff
@@ -261,11 +263,50 @@ where
     #[tracing::instrument(skip(self))]
     async fn get_pending_requests(
         &mut self,
+        bitcoin_chain_tip: &model::BitcoinBlockHash,
         signer_btc_state: utxo::SignerBtcState,
     ) -> Result<utxo::SbtcRequests, error::Error> {
-        // TODO(318): Fetch pending deposit and withdraw requests
-        // with enough votes and connect them into the right format
-        todo!();
+        let context_window = self
+            .context_window
+            .try_into()
+            .map_err(|_| error::Error::TypeConversion)?;
+
+        let threshold = self.threshold.into();
+
+        let pending_deposit_requests = self
+            .storage
+            .get_pending_accepted_deposit_requests(bitcoin_chain_tip, context_window, threshold)
+            .await?;
+
+        let pending_withdraw_requests = self
+            .storage
+            .get_pending_accepted_withdraw_requests(bitcoin_chain_tip, context_window, threshold)
+            .await?;
+
+        let deposits: Vec<utxo::DepositRequest> = pending_deposit_requests
+            .into_iter()
+            .map(|request| request.try_into())
+            .collect::<Result<_, _>>()?;
+
+        let withdrawals = pending_withdraw_requests
+            .into_iter()
+            .map(|request| request.try_into())
+            .collect::<Result<_, _>>()?;
+
+        let accept_threshold = self.threshold;
+        let num_signers = self
```

Comment:
Hmmm, maybe I should change `num_signers` to a `usize` in `utxo::SbtcRequests`? Eh, it that change looks annoying.